evap/middleware.py

```diff
@@ -26,3 +28,21 @@ def process_view(request, view_func, _view_args, _view_kwargs):
 def no_login_required(func):
     func.no_login_required = True
     return func
+
+
+def user_language_middleware(get_response):
+    def middleware(request):
+        if not (request.user and request.user.is_authenticated):
+            return get_response(request)
+
+        if request.user.language:
+            translation.activate(request.user.language)
+        else:
+            request.user.language = translation.get_language()
+            request.user.save()
+        lang = request.user.language
+        response = get_response(request)
+        response.set_cookie(settings.LANGUAGE_COOKIE_NAME, lang)
+        return response
```

Comment:
- The average user will be logged in and authenticated, right? With this code, they will always have a `user.language` set except for their very first request after account creation, right? If that is the case, I would add a comment to that branch explaining that
- For the average request, we would call `translation.activate` and `set_cookie` here, right? Feels a bit heavy to me, and I think it's weird that we would need a custom middleware. But if that is what it takes to make it work, then that's just the way it is ¯\_(ツ)_/¯
- Setting the cookie is only necessary so the language is kept after the user logs out, right? In that case, I would also add a comment.

This will sync the languages between all devices, so a language change on one device will affect all other devices, right? @janno42 is that what we want?